File: sunflower-gin-vue-master/sunflower-gin/api/checkin/v1/checkin.go
package v1

type DailyResp struct{}

// CalendarReq 签到日历请求参数
type CalendarReq struct {
	YearMonth string `form:"yearMonth" binding:"required"` // ?yearMonth=2025-01
}

// CalendarResp 签到日历响应参数
type CalendarResp struct {
	Year   int        `json:"year"`
	Month  int        `json:"month"`
	Detail DetailInfo `json:"detail"`
}

type DetailInfo struct {
	CheckedInDays      []int `json:"checkedInDays"`      // 签到的日期序号
	RetroCheckedInDays []int `json:"retroCheckedInDays"` // 补签的日期序号
	IsCheckedInToday   bool  `json:"isCheckedInToday"`   // 今天是否签到
	RemainRetroTimes   int   `json:"remainRetroTimes"`   // 剩余补签次数
	ConsecutiveDays    int   `json:"consecutiveDays"`    // 连续签到天数
}

// RetroReq 补签请求参数
type RetroReq struct {
	Date string `json:"date" binding:"required"`
}

// RetroResp 补签响应参数
type RetroResp struct{}

File: SignEase-gin-vue-/frontend/src/stores/checkin.ts
// 签到数据状态管理
import { defineStore } from 'pinia'
import { ref } from 'vue'
import { getPointsInfo, getCheckinCalendarDetail, clearCache } from '@/api/adapter'
import type { PointsInfo, CheckInCalendarDetail } from '@/types'

export const useCheckinStore = defineStore('checkin', () => {
  // 状态
  const pointsInfo = ref<PointsInfo>({
    totalPoints: 0,
    consecutiveDays: 0,
    retroAvailable: 0,
    checkedInToday: false,
    retroCheckedInDays: [],
  })

  const calendarDetail = ref<CheckInCalendarDetail>({
    checkedInDays: [],
    retroCheckedInDays: [],
    isCheckedInToday: false,
    remainRetroTimes: 0,
    consecutiveDays: 0,
  })

  const loading = ref(false)
  const error = ref('')

  // 获取积分信息
  async function fetchPointsInfo() {
    if (loading.value) return // 防止重复请求

    loading.value = true
    error.value = ''
    try {
      const data = await getPointsInfo()
      pointsInfo.value = data
      console.log('积分信息获取成功:', data)
    } catch (err: unknown) {
      console.error('获取积分信息失败', err)
      error.value = err instanceof Error ? err.message : '获取积分信息失败'
      // 不要抛出错误，让组件继续正常工作
    } finally {
      loading.value = false
    }
  }

  // 获取签到日历详情
  async function fetchCalendarDetail(year: number, month: number) {
    if (loading.value) return // 防止重复请求

    loading.value = true
    error.value = ''
    try {
      const response = await getCheckinCalendarDetail(year, month)
      calendarDetail.value = response.detail

      // 同步更新积分信息中的相关字段
      pointsInfo.value.consecutiveDays = response.detail.consecutiveDays
      pointsInfo.value.retroAvailable = response.detail.remainRetroTimes
      pointsInfo.value.checkedInToday = response.detail.isCheckedInToday
      pointsInfo.value.retroCheckedInDays = response.detail.retroCheckedInDays
    } catch (err: unknown) {
      console.error('获取签到日历详情失败', err)
      error.value = err instanceof Error ? err.message : '获取签到日历详情失败'
    } finally {
      loading.value = false
    }
  }

  // 刷新所有数据
  async function refreshData(year?: number, month?: number) {
    clearCache() // 清除缓存
    await fetchPointsInfo()
    if (year !== undefined && month !== undefined) {
      await fetchCalendarDetail(year, month)
    }
  }

  // 重置状态
  function resetState() {
    pointsInfo.value = {
      totalPoints: 0,
      consecutiveDays: 0,
      retroAvailable: 0,
      checkedInToday: false,
      retroCheckedInDays: [],
    }
    calendarDetail.value = {
      checkedInDays: [],
      retroCheckedInDays: [],
      isCheckedInToday: false,
      remainRetroTimes: 0,
      consecutiveDays: 0,
    }
    error.value = ''
  }

  return {
    pointsInfo,
    calendarDetail,
    loading,
    error,
    fetchPointsInfo,
    fetchCalendarDetail,
    refreshData,
    resetState,
  }
})
